OOP/LibraryManagement/LibraryApp/Repositories/JsonFileDocumentRepository.cs

```diff
@@ -0,0 +1,99 @@
+﻿using LibraryApp.Documents;
+using Newtonsoft.Json.Linq;
+using System.Text.RegularExpressions;
+using System.Text.Json;
+using LibraryApp.Services;
+
+namespace LibraryApp.Repositories;
+
+public class JsonFileDocumentRepository<T>(DocumentCache cache) : IRepository<T> where T : Document
+{
+    private const string DocType = "documentType";
+    private readonly DocumentCache _cache = cache;
+    private readonly string _baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+    private readonly JsonSerializerOptions _jsonOptions = new() { WriteIndented = true, IncludeFields = true };
+
+    public T Read(string title)
+    {
+        var document = _cache.GetDocument<T>(title);
+
+        if (document == null)
+        {
+            var file = new DirectoryInfo(_baseDirectory).GetFiles().FirstOrDefault(f => f.Name.Contains(title));
+
+            if (file != null)
+            {
+                var json = File.ReadAllText(file.FullName);
+                var jObject = JObject.Parse(json);
+                var docType = jObject[DocType].ToObject<DocumentType>();
+
+                document = Deserialize(jObject, docType);
+                _cache.SetDocument(document);
+            }
+        }
+
+        return document;
+    }
+
+    public IList<T> ReadAll()
+    {
+        var directoryInfo = new DirectoryInfo(_baseDirectory);
+        var files = directoryInfo.GetFiles("type_*_*_*.json");
+        var documents = new List<T>();
+
+        foreach (var file in files)
+        {
+            var title = GetTitle(file.Name);
+            var document = Read(title);
+
+            if (document != null)
+            {
+                documents.Add(document);
+            }
+        }
+
+        return documents;
+    }
+
+    public void Create(T document)
+    {
+        if (Read(document.Title) != null)
```

Comment:
If you want just to check that file exists or not I would advise to use corresponding method - [`Path.Exists`](https://www.geeksforgeeks.org/file-exists-method-in-c-sharp-with-examples/) instead of uploading. The reason is that it will require additional resources allocation that you ain't gonna use. Let's, please, add instead some private bool method that checks for file existence.